scripts/run_performance_job.py

```diff
@@ -50,258 +50,613 @@ def output_counters_for_crank(reports: list[Any]):
+    # Add the install_prerequisites to the pre_commands
+    if args.os_group == "windows":
+        # TODO: Should we also give Windows the same treatment as linux and ensure that each command succeeds?
+        helix_pre_commands += install_prerequisites
+    else:
+        if install_prerequisites:
+            combined_prequisites = " && ".join(install_prerequisites)
+            helix_pre_commands += [
                 'echo "** Installing prerequistes **"',
-                ("python3 -m pip install --user -U pip && "
-                "sudo apt-get -y install python3-venv && "
-                "python3 -m venv $HELIX_WORKITEM_PAYLOAD/.venv && "
-                "ls -l $HELIX_WORKITEM_PAYLOAD/.venv/bin/activate && "
-                "export PYTHONPATH= && "
-                "python3 -m pip install --user -U pip && "
-                "pip3 install --user urllib3==1.26.18 && "
-                "pip3 install --user azure.storage.blob==12.0.0 && "
-                "pip3 install --user azure.storage.queue==12.0.0 && "
-                "sudo apt-get update && "
-                "sudo apt -y install curl dirmngr apt-transport-https lsb-release ca-certificates && "
-                f"{wasm_precommand} && "
-                f"export PERFLAB_UPLOAD_TOKEN=\"{args.perflab_upload_token}\" "
-                "|| export PERF_PREREQS_INSTALL_FAILED=1"),
-            'test "x$PERF_PREREQS_INSTALL_FAILED" = "x1" && echo "** Error: Failed to install prerequites **"'
+                f"{combined_prequisites} || export PERF_PREREQS_INSTALL_FAILED=1",
+                'test "x$PERF_PREREQS_INSTALL_FAILED" = "x1" && echo "** Error: Failed to install prerequites **"'
             ]
 
-    mono_interpreter = False
+    # Set MONO_ENV_OPTIONS with for Mono Interpreter runs
     if args.codegen_type == "Interpreter" and args.runtime_type == "mono":
-        mono_interpreter = True
         if args.os_group == "windows":
             helix_pre_commands += ['set MONO_ENV_OPTIONS="--interpreter"']
         else:
             helix_pre_commands += ['export MONO_ENV_OPTIONS="--interpreter"']
 
+    # Enable MSBuild node communication logs
     if args.os_group == "windows":
         helix_pre_commands += ["set MSBUILDDEBUGCOMM=1", 'set "MSBUILDDEBUGPATH=%HELIX_WORKITEM_UPLOAD_ROOT%"']
-        helix_post_commands = ["set PYTHONPATH=%ORIGPYPATH%"]
     else:
         helix_pre_commands += ["export MSBUILDDEBUGCOMM=1", 'export "MSBUILDDEBUGPATH=$HELIX_WORKITEM_UPLOAD_ROOT"']
+
+    # Copy the performance repo and root directory to the work item directory
+    if args.os_group == "windows":
+        helix_pre_commands += [ 
+            "robocopy /np /nfl /e %HELIX_CORRELATION_PAYLOAD%\\performance %HELIX_WORKITEM_ROOT%\\performance",
+            "robocopy /np /nfl /e %HELIX_CORRELATION_PAYLOAD%\\root %HELIX_WORKITEM_ROOT%" ]
+    else:
+        helix_pre_commands += [ 
+            "cp -R $HELIX_CORRELATION_PAYLOAD/performance/* $HELIX_WORKITEM_ROOT/performance",
```

Comment:
This line seems to be failing at least in the ubuntu main akadeindex runs:
```
+ cp -R /datadisks/disk1/work/AFFE096D/p/performance/CODE-OF-CONDUCT.md /datadisks/disk1/work/AFFE096D/p/performance/Directory.Build.props /datadisks/disk1/work/AFFE096D/p/performance/Directory.Build.targets /datadisks/disk1/work/AFFE096D/p/performance/LICENSE.TXT /datadisks/disk1/work/AFFE096D/p/performance/NuGet.config /datadisks/disk1/work/AFFE096D/p/performance/README.md /datadisks/disk1/work/AFFE096D/p/performance/azure-pipelines.yml /datadisks/disk1/work/AFFE096D/p/performance/docs /datadisks/disk1/work/AFFE096D/p/performance/eng /datadisks/disk1/work/AFFE096D/p/performance/global.json /datadisks/disk1/work/AFFE096D/p/performance/global.net8.json /datadisks/disk1/work/AFFE096D/p/performance/helix.yml /datadisks/disk1/work/AFFE096D/p/performance/requirements.txt /datadisks/disk1/work/AFFE096D/p/performance/scripts /datadisks/disk1/work/AFFE096D/p/performance/src /datadisks/disk1/work/AFFE096D/p/performance/tools /datadisks/disk1/work/AFFE096D/w/B65309F1/e/performance
cp: target '/datadisks/disk1/work/AFFE096D/w/B65309F1/e/performance' is not a directory
```

The runs still seems to succeed though. Looking at the logs and the directories for everything, is there a reason we try to copy this stuff to the HELIX_WORKITEM_ROOT? We seem to use the HELIX_WORKITEM_ROOT in the python path and it seems to be the default working directory, but all our other calls use the HELIX_CORRELATION_PAYLOAD for running.

Reply:
This is not something we have to decide on for this PR, but we should double check this logic and can potentially simplify part of our pipeline in the near future.

Reply:
Digging deeper, it looks like this originally comes from helix.proj and was only added when UseCoreRun was true. It may be worth moving or wrapping these file copies with a check for UseCoreRun for consistency, although I don't think it will directly affect the performance runs if we keep it aside from the failure to copy.

It also seems that the additional /* is causing the command to expand rather than using `/performance/*`, which combined with not having a / at the end of the target directory is causing the error. An example from the dotnet-runtime-perf repo of the call is `+ cp -R <PAYLOAD>/performance <WORKITEM>/performance`.

Reply:
This change is actually something new I am introducing as part of this script, and is now required for all runs, not just corerun runs. I had forgotten to mention it in the original post. 

Every helix job has one work item directory (which Helix will upload and then copy into the working directory on the host machine), and N correlation payload directories. My understanding is that when we first started out, we always just made the work item directory equal to the performance directory. Then eventually we found that this caused issues on Windows because we would be logging inside the performance directory at the same time as we were uploading it, which caused the helix publish to hang. As a result, we made it so that the performance directory would be copied to another folder, "notLocked" and then once it gets into Helix it would be copied back to the work item root again. Similarly, in the runtime repository it also does something similar except it has to upload some additional correlation payloads for coreroot. With this change, I've made it so that the performance repository will always be copied into `CorrelationPayload/performance` on the local machine, which will then get uploaded to a `performance` directory in Helix, and then this line will copy it back into the work item root.

As for the issue you raised, it turns out that even though I attempted to port the code that would copy the correlation payload to the work item root, I had left the code that referenced the performance repository still pointing to the helix correlation payload. I have just pushed a fix now that ensures that the target directory in the work item root exists and that it actually gets called from there. I think that even though it did work without this copy happening, it still probably needs to happen so that the custom nuget config that we copy into the work item root gets referenced when the project is built as part of the benchmark.